### PR TITLE
Feature/refactor code

### PR DESCRIPTION
**Pull request by Ninebell**

### CODE REVIEW

File: dataset/compressed.py
Comment (inqlee0704) [inline dataset/compressed.py:1]: 얘는 파일이름 encodeddataset 으로 안바꿔도 괜찮을까여??

Comment (Ninebell) [reply dataset/compressed.py:1]: encodeddataset 너무 복잡한거같아서 단순화 시켰어요

File: train_val.py [hunk 1]
```diff
@@ -179,132 +80,96 @@ def main():
...
-        best_loss = checkpoint["loss"]
-        print(f"Loaded checkpoint from epoch {start_epoch}")
-    else:
-        start_epoch = 0
-        best_loss = 1e10
+    valid_dataset = EncodedDataset(data_root, data_aug=False, mode="valid")
+    valid_loader = DataLoader(valid_dataset, batch_size=batch_size, shuffle=False)
 
     train_loss_list = []
     valid_loss_list = []
```
Comment (inqlee0704) [inline train_val.py]: train_loss_list 랑 valid_loss_list 안쓰는데 이거 지울까여?

File: train_val.py [hunk 2]
```diff
@@ -179,132 +80,96 @@ def main():
...
             optimizer.step()
-            optimizer.zero_grad()
             pbar.set_description(
                 f"Epoch {epoch} at {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}, loss: {loss.item():.4f}"
             )
+            # break
         epoch_end_time = time()
         train_epoch_loss = train_epoch_loss / len(train_loader)
 
         train_loss_list.append(train_epoch_loss)
```
Comment (inqlee0704) [inline train_val.py]: 이거도 그냥 지우면 될듯 합니다

File: train_val.py [hunk 3]
```diff
@@ -198,113 +75,105 @@ def main():
...
+    lr_scheduler = optim.lr_scheduler.ReduceLROnPlateau(
+        optimizer, mode="min", factor=0.9, patience=5, cooldown=0
+    )
+
+    # * Dataloader
+    train_dataset = EncodedDataset(data_root, data_aug=True, mode="train")
+    train_loader = DataLoader(train_dataset, batch_size=batch_size, shuffle=True)
+
+    valid_dataset = EncodedDataset(data_root, data_aug=False, mode="valid")
+    valid_loader = DataLoader(valid_dataset, batch_size=batch_size, shuffle=False)
```
Comment (inqlee0704) [inline train_val.py]: valid_loader = DataLoader(valid_dataset, batch_size=batch_size*2, shuffle=False)
valid는 backpropagation없어서 배치 두배해도 train 메모리보다 조금 더 먹어서, 방금 2배로 돌려봤는데, 4090에서 돌아가고, valid 속도 약 두배 빨라지네여 ㅋㅋㅋㅋ
